microprofile/grpc/metrics/src/main/java/io/helidon/microprofile/grpc/metrics/GrpcMetricRegistrationObserverImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.grpc.metrics;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.helidon.microprofile.metrics.MetricAnnotationDiscovery;
+import io.helidon.microprofile.metrics.spi.MetricRegistrationObserver;
+
+import org.eclipse.microprofile.metrics.Metadata;
+import org.eclipse.microprofile.metrics.Metric;
+import org.eclipse.microprofile.metrics.MetricID;
+
+/**
+ * The gRPC implementation of {@link io.helidon.microprofile.metrics.spi.MetricRegistrationObserver} with a static factory method.
+ */
+public class GrpcMetricRegistrationObserverImpl implements MetricRegistrationObserver {
+
+    private static GrpcMetricRegistrationObserverImpl instance;
+
+    static GrpcMetricRegistrationObserverImpl instance() {
```

Comment:
Same as with previous type - static instance that is stored from constuctor is unusual and I am not sure what it is trying to achieve

Reply:
Both of these types of observers need to be singletons. The metrics CDI extension uses service loading to obtain each one and store them for later use. Later, gRPC needs a way to find the already-created singletons. 

Storing the instance statically makes it easy for gRPC to find the singletons.

Another option I considered: The metrics CDI could expose methods which let gRPC explicitly enroll its observers during `BeforeBeanDiscovery`. The gRPC extension would instantiate its observers however it wanted, use CDI to look up the metrics extension, and pass the observers to the metrics extension `enroll` methods. I wanted to avoid exposing yet another public method on the metrics extension. 

Would that approach resolve your concerns?